src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -110,24 +110,29 @@ def download(
 
 
 def _get_files(
-    file_name: Union[str, pathlib.PurePath, list[Union[str, pathlib.PurePath]]]
+    file_name: Union[str, pathlib.PurePath, list[Union[str, pathlib.PurePath]]],
+    path: Optional[str],
 ):
-    path = "/home/runner/.local/share/ansys_fluent_core/examples"
-    if os.path.exists(path):
-        if isinstance(file_name, str):
-            files = [str(os.path.join(path, os.path.basename(file_name)))]
-        elif isinstance(file_name, pathlib.PurePath):
-            files = [str(os.path.join(path, file_name.name))]
-        elif isinstance(file_name, list):
+    if isinstance(file_name, str):
+        files = (
+            [str(os.path.join(path, os.path.basename(file_name)))]
+            if os.path.isfile(os.path.join(path, os.path.basename(file_name)))
+            else [file_name]
+        )
+    elif isinstance(file_name, pathlib.PurePath):
+        files = (
+            [str(os.path.join(path, file_name.name))]
+            if os.path.isfile(os.path.join(path, file_name.name))
+            else [str(file_name)]
+        )
+    elif isinstance(file_name, list):
+        if os.path.exists(path):
```

Comment:
What is going on here? Seems unrelated to the changes described in the PR. So `path` was hardcoded to work only in GitHub runners previously?

Also, `pathlib.Path` should be able to handle `str` paths with no issues in my understanding, do we need two different `if` clauses for `str` and `Path`?

Lastly, usually we don't need to convert a `pathlib.Path` to `str` as in line 126, in my understanding most Python tools should work with `pathlib.Path` out of the box (including all built-in functions like `open`).

Reply:
There was a bug in locating files in GitHub (container mode) as the path was hard-coded. We have resolved this now on the basis of Fluent mode. If the Fluent is running is container mode then it will check `host_mount_path` as `path` to locate files if that path is already exists else it will use given file name directly.

Reply:
@raph-luc 

Yes, I verified that we don't need to convert `pathlib.Path` to `str`. We can use it directly. We will remove `str` for `pathlib.Path`.  Thank you.

```
>>> file_path = pathlib.Path("D:/Repos/pyfluent/bug_solver.cas.h5")                            
>>> file_path                                                                                  
WindowsPath('D:/Repos/pyfluent/bug_solver.cas.h5')
>>> from ansys.fluent.core import launch_fluent 
>>> solver = launch_fluent()
>>> solver.file.read_case(file_name=file_path)
pyfluent.settings_api WARNING: Mismatch between generated file and server object info. Dynamically created settings classes will be used.
Fast-loading "D:\ANSYSDev\vNNN\fluent\fluent24.2\\addons\afd\lib\hdfio.bin"
Done.
Multicore SMT processors detected. Processor affinity set!

Reading from AAPQ13pUvWxQtxN:"D:\Repos\pyfluent\bug_solver.cas.h5" in NODE0 mode ...
  Reading mesh ...
       17822 cells,     1 cell zone  ...
          17822 mixed cells,  zone id: 87
       91581 faces,     7 face zones ...
           2168 polygonal wall faces,  zone id: 34
            268 polygonal wall faces,  zone id: 33
            155 polygonal pressure-outlet faces,  zone id: 32
            152 polygonal velocity-inlet faces,  zone id: 31
             55 polygonal velocity-inlet faces,  zone id: 30
           2001 polygonal symmetry faces,  zone id: 29
          86782 mixed interior faces,  zone id: 89
       66417 nodes,     1 node zone  ...
  Done.


Building...
     mesh
     materials,
     interface,
     domains,
        mixture
     zones,
        wall-elbow
        wall-inlet
        outlet
        cold-inlet
        hot-inlet
        symmetry-xyplane
        interior--elbow-fluid
        elbow-fluid
     parallel,
Done.
>>> 
```

Reply:
Done. Thank you.